crates/extension_host/src/wasm_host/wit/since_v0_8_0.rs

```diff
@@ -1108,4 +1108,45 @@ impl ExtensionImports for WasmState {
             .with_context(|| format!("setting permissions for path {path:?}"))
             .to_wasmtime_result()
     }
+
+    async fn log_trace(&mut self, message: String) -> wasmtime::Result<()> {
+        let logger = get_logger(&self.manifest.id);
+        zlog::trace!(logger => "{:?}", message);
+
+        Ok(())
+    }
+
+    async fn log_debug(&mut self, message: String) -> wasmtime::Result<()> {
+        let logger = get_logger(&self.manifest.id);
+        zlog::debug!(logger => "{:?}", message);
+
+        Ok(())
+    }
+
+    async fn log_info(&mut self, message: String) -> wasmtime::Result<()> {
+        let logger = get_logger(&self.manifest.id);
+        zlog::info!(logger => "{:?}", message);
+
+        Ok(())
+    }
+
+    async fn log_warn(&mut self, message: String) -> wasmtime::Result<()> {
+        let logger = get_logger(&self.manifest.id);
+        zlog::warn!(logger => "{:?}", message);
+
+        Ok(())
+    }
+
+    async fn log_error(&mut self, message: String) -> wasmtime::Result<()> {
+        let logger = get_logger(&self.manifest.id);
+        zlog::error!(logger => "{:?}", message);
+
+        Ok(())
+    }
+}
+
+fn get_logger(extension_id: &Arc<str>) -> zlog::Logger {
+    let all_extensions_logger = zlog::scoped!("extension");
+    let current_extension_scope = Box::leak(Box::new(extension_id.to_string()));
```

Comment:
`zlog:scoped!` requires an `&'static str`. This is the only way I could find to get that.

I think that `Box::leak` is fine here because the logger will need to remain until the program terminates in most cases (an exception being if the extension is disabled).

I don't know if calling `Box::leak(Box::new(extension_id.to_string()))` multiple times with the same extension id leaks a single string or if it continues to leak a new one each time. If it's a new one each time, then something about this would need to be changed.